Support RegularExpression for path matching

dmavrommatis opened the pull request:
<!--
PLEASE READ THIS MESSAGE.

Documentation fixes or enhancements:
- for Traefik v2: use branch v2.11
- for Traefik v3: use branch v3.0

Bug fixes:
- for Traefik v2: use branch v2.11
- for Traefik v3: use branch v3.0

Enhancements:
- for Traefik v2: we only accept bug fixes
- for Traefik v3: use branch v3.0

HOW TO WRITE A GOOD PULL REQUEST? https://doc.traefik.io/traefik/contributing/submitting-pull-requests/

-->

### What does this PR do?

<!-- A brief description of the change being made with this pull request. -->
Pull request introduced support for path matching of type `RegularExpression` from Gateway API.

### Motivation

<!-- What inspired you to submit this pull request? -->
Moving from traefik/IngressRoute to gateway/HttpRoute is not possible since we are using a lot of `PathRegexp` which is not yet supported.

### More

- [x] Added/updated tests
- [ ] Added/updated documentation

### Additional Notes

<!-- Anything else we should know when reviewing? -->

kevinpollet commented:
Hello @dmavrommatis and thanks for your contribution,

As this is an enhancement, please could you rebase this pull request on master?

dmavrommatis commented:
> Hello @dmavrommatis and thanks for your contribution,
> 
> As this is an enhancement, please could you rebase this pull request on master?

Changed the target branch to master and rebased. Previously I chose v3.0 branch because of the PR guide advising to use the v3.0 branch for enchancements.
```
Enhancements:
- for Traefik v2: we only accept bug fixes
- for Traefik v3: use branch v3.0
```

kevinpollet reviewed (approved):
Thanks 👍

rtribotte reviewed (approved):
Thanks!

lbenguigui reviewed (approved):
LGTM
